### PR TITLE
Add post "Migrating Existing Datastores"

### PR DESCRIPTION
**Pull request by jiahaog**

### CODE REVIEW

File: _posts/2017-08-08-migrating-existing-datastores.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+id: migrating-existing-datastores
+title: "Migrating Existing Datastores"
```
Comment (yangshun) [inline _posts/2017-08-08-migrating-existing-datastores.md]: I think from the Google doc, the title is "So you are plan to migrate from existing datastore. Here's what comes next", unless you all agreed to change it?

File: _posts/2017-08-08-migrating-existing-datastores.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+At Grab we take pride in creating solutions that impact millions of people in Southeast Asia and as they say, with great power comes great responsibility. As an app with 45 million downloads and 1.1 million drivers, it's our responsibility to keep our systems up-and-running. Any downtime causes drivers to miss earning and passengers to miss their appointments.
+
+It all started when in early 2017, Grab Identity team realised that given the rate at which our user base was growing, we wouldn't be able to sustain the load with our existing single Redis node architecture. We used Redis as a cache to store authentication tokens required for secure mobile client to server communication. These tokens are permanently backed up in an underlying MySQL store. The existing Redis instance was filling at crazy speeds and we were growing at a rate at which we had a maximum of 2 months to react before we would start to ‘choke’ i.e. running out of memory to store more data or run operations on the above mentioned Redis node.
+
+![image1](/img/migrating-existing-datastores/image1.png)
```
Comment (yangshun) [inline _posts/2017-08-08-migrating-existing-datastores.md]: Need to give better names for the images.

File: _posts/2017-08-08-migrating-existing-datastores.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+From all the mentioned options, it was clear to us that achieving a completely horizontally scalable model where data-sources could be increased on demand with ease, was not possible with the Redis-AWS combination (unless we ended up with a [self-hosted Redis](https://redis.io/topics/cluster-spec) on EC2). This is when we started questioning some assumptions:
+
+Did we need horizontal scalability for all the operations?
+
+And we had the answer to this. In a typical authentication system, the scale of writes is significantly lower compared to that of reads. A token that was provisioned in 1 request, would end up being used to authenticate another N requests and our graphs validated this:
+
+Write load
+
+![image2](/img/migrating-existing-datastores/image2.png)
```
Comment (yangshun) [inline _posts/2017-08-08-migrating-existing-datastores.md]: naming

File: _data/authors.yml [hunk 4]
```diff
@@ -42,6 +42,9 @@ lian-yuanlin:
   name: Lian Yuanlin
   thumbnail: /img/authors/lian-yuanlin.jpg
   github: alienchow
+nishant-gupta:
+  name: Nishant Gupta
+  thumbnail: /img/authors/nishant-gupta.jpg
```
Comment (yangshun) [inline _data/authors.yml]: How big is the photo? Generally I try to resize to 256 x 256 size and compress first.

File: _posts/2017-08-08-migrating-existing-datastores.md (hunk 1 above)
Comment (jiahaog) [inline _posts/2017-08-08-migrating-existing-datastores.md]: Spoke to Nishant and this is okay with him

File: _data/authors.yml (hunk 4 above)
Comment (jiahaog) [inline _data/authors.yml]: Yup it's 256 x 256

**Pull request merged**